<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ONNXProfiler</class>
 <widget class="QMainWindow" name="ONNXProfiler">
  <property name="geometry">
   <rect>
    <x>0</x><y>0</y><width>1000</width><height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ONNX Model Profiler</string>
  </property>

  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="mainLayout">

    <!-- Left side -->
    <item>
     <layout class="QVBoxLayout" name="leftLayout">
      <item><widget class="QLineEdit" name="folder_input"/></item>
      <item>
       <layout class="QHBoxLayout" name="button_layout">
        <item>
         <widget class="QPushButton" name="browse_button">
          <property name="text"><string>Select Folder...</string></property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="profile_button">
          <property name="text"><string>Profile</string></property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTreeView" name="model_tree_view">
        <property name="minimumSize">
         <size><width>300</width><height>300</height></size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="generate_button">
        <property name="text"><string>Generate Model Deployment Code</string></property>
       </widget>
      </item>
     </layout>
    </item>

    <!-- Right side -->
    <item>
     <widget class="QSplitter" name="right_splitter">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>

      <!-- Tabs -->
      <widget class="QTabWidget" name="result_tab_widget">
       <property name="minimumSize">
        <size><width>500</width><height>400</height></size>
       </property>

       <widget class="QWidget" name="cpu_tab">
        <attribute name="title"><string>CPU</string></attribute>
        <layout class="QVBoxLayout" name="cpu_layout">
         <item>
          <widget class="QTableWidget" name="cpu_table"/>
         </item>
        </layout>
       </widget>

       <widget class="QWidget" name="npu1_tab">
        <attribute name="title"><string>NPU1</string></attribute>
        <layout class="QVBoxLayout" name="npu1_layout">
         <item>
          <widget class="QTableWidget" name="npu1_table"/>
         </item>
        </layout>
       </widget>

       <widget class="QWidget" name="npu2_tab">
        <attribute name="title"><string>NPU2</string></attribute>
        <layout class="QVBoxLayout" name="npu2_layout">
         <item>
          <widget class="QTableWidget" name="npu2_table"/>
         </item>
        </layout>
       </widget>
      </widget>

      <!-- Log Output -->
      <widget class="QPlainTextEdit" name="log_output">
       <property name="readOnly"><bool>true</bool></property>
       <property name="placeholderText"><string>Log Output Area</string></property>
      </widget>
     </widget>
    </item>

   </layout>
  </widget>

  <widget class="QMenuBar" name="menubar"/>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
